src/cogent3/core/new_sequence.py

```diff
@@ -1277,36 +1276,39 @@ def with_masked_annotations(
         return new
 
     def gapped_by_map_segment_iter(
-        self, segment_map, allow_gaps=True, recode_gaps=False
+        self, map, allow_gaps=True, recode_gaps=False
```

Comment:
If this is on a sequence, shouldn't it be returning an instance of self?

Reply:
it iters the segments as str

Reply:
the method below`gapped_by_map` return an instance of self